### PR TITLE
add new tests for ES{256,384,512}

### PR DESCRIPTION
**Pull request by thomas-fossati**
Also a bunch of fixes and some cleanup.

Partially addresses #1

### CODE REVIEW

File: sign1-sign-0000.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "uuid": "D55A49BD-53D9-42B1-9E76-E0CF2AD33E9D",
-  "title": "And another one...",
-  "description": "Second attempt at a Sign1 test case (sign)",
-   "key": {
+  "title": "Sign1 w/ external input - ECDSA w/ SHA-256 (sign)",
+  "description": "Sign with one signer using ECDSA w/ SHA-256 supplying external input",
```
Comment (yogeshbdeshpande) [inline sign1-sign-0000.json:4]: Isn't Sign1 an explicit indication that it is with one signer? Should you want to retain one signer in the description, I am ok with that so long as you make it consistent across all test descriptions below!

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Comment by qmuntal** [comment]
I've successfully used these new tests in https://github.com/veraison/go-cose/pull/19.

Just one comment: Verify cases don't need private key parameters. I would remove them to avoid confusion.

**Comment by thomas-fossati** [comment]
Hearing no dissenting voices I'm going to merge this PR.

**Pull request merged**